aws/service.go

```diff
@@ -1737,6 +1739,10 @@ func getSessionWithMaxRetries(ctx context.Context, d *plugin.QueryData, region s
 
 	if awsConfig.Profile != nil {
 		sessionOptions.Profile = *awsConfig.Profile
+		// When profile is set but not access or secret key, logically this means it will be SSO login
```

Comment:
@dbmurphy Is this always true? If I have a connection using my `default` profile (or any other named profile):
```
connection "aws" {
    plugin = "aws"
    regions = ["us-east-1"]
}
```

Then have my `~/.aws/credentials` file setup so my `default` profile uses an AWS access key pair:
```
[default]
aws_access_key_id = AKIA...
aws_secret_access_key = cD+J...
```

The `awsConfig.AccessKey` and `awsConfig.SecretKey` values will still be `nil`, as they're not explicitly defined in the config file. Similarly, there are a few other ways you can authenticate with the AWS plugin that do not require an access or secret key explicitly set in the Steampipe AWS config file, like using AssumeRole creds, AWS Vault creds, and EC2 instance profile creds. There are a few examples of each sample connection configuration from https://hub.steampipe.io/plugins/turbot/aws.

Have you tried this code while using any of the above authentication methods?

Reply:
> @dbmurphy Is this always true? If I have a connection using my `default` profile (or any other named profile):
> 
> ```
> connection "aws" {
>     plugin = "aws"
>     regions = ["us-east-1"]
> }
> ```
> 
> Then have my `~/.aws/credentials` file setup so my `default` profile uses an AWS access key pair:
> 
> ```
> [default]
> aws_access_key_id = AKIA...
> aws_secret_access_key = cD+J...
> ```
> 
> The `awsConfig.AccessKey` and `awsConfig.SecretKey` values will still be `nil`, as they're not explicitly defined in the config file. Similarly, there are a few other ways you can authenticate with the AWS plugin that do not require an access or secret key explicitly set in the Steampipe AWS config file, like using AssumeRole creds, AWS Vault creds, and EC2 instance profile creds. There are a few examples of each sample connection configuration from https://hub.steampipe.io/plugins/turbot/aws.
> 
> Have you tried this code while using any of the above authentication methods?

I have tried some but not all, this is rather complicated, and why I made mention that the auth system really needs some TLC, so we can do things like 

if isSSO
if isKey
...

However this means you would need to refactor the Env, SPC, and .aws layer to get a materialized view of what the SDK itself would see. In the current form of this code, it's possible if you had a key type auth and it tried this is would call sts and sso butt hey would fail as no SSO URL is defined in the .aws/config. In this case, it would simply continue and the SDK would still use the key/secret you have provided. 

As such, I feel this is a solution that works but could be made cleaner, but such clean is a major refactor of the auth code. Similar to how it should only call SSO login/sts checks once but as steampipe does not have a hierarchical order and all copies of the AWS plugin as started at the same time it calls N open browser or log prints for authorizing in your SSO system.